[COSMETIC]
I was not to happy with the previous patch, because fontset got NULL. I think
it's better, if in this case the old font would remain and only an error is
returned.

--- src/libxosd/xosd.c~	2003-09-29 08:32:20.000000000 +0200
+++ src/libxosd/xosd.c	2003-09-29 08:31:36.000000000 +0200
@@ -559,6 +559,7 @@ force_redraw (xosd * osd, int line)	/* R
 static int
 set_font (xosd * osd, const char *font)	/* Requires mutex lock. */
 {
+  XFontSet fontset2;
   char **missing;
   int nmissing;
   char *defstr;
@@ -567,20 +568,23 @@ set_font (xosd * osd, const char *font)	
   XFontSetExtents *extents;
 
   assert (osd);
-  if (osd->fontset)
-    {
-      XFreeFontSet (osd->display, osd->fontset);
-      osd->fontset = NULL;
-    }
 
-  osd->fontset =
+  /* Try to create the new font. If it doesn't succeed, keep old font. */
+  fontset2 =
     XCreateFontSet (osd->display, font, &missing, &nmissing, &defstr);
-  if (osd->fontset == NULL)
+  XFreeStringList (missing);
+  if (fontset2 == NULL)
     {
       xosd_error = "Requested font not found";
       return -1;
     }
-  XFreeStringList (missing);
+
+  if (osd->fontset)
+    {
+      XFreeFontSet (osd->display, osd->fontset);
+      osd->fontset = NULL;
+    }
+  osd->fontset = fontset2;
 
   extents = XExtentsOfFontSet (osd->fontset);
   osd->extent = &extents->max_logical_extent;
